src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1231,6 +1267,12 @@ internal async Task HandleAgentNotebooksRequest(AgentNotebooksParams parameters,
             {
                 result.Success = false;
                 result.ErrorMessage = e.ToString();
+                Exception exception = e.InnerException;
+                while (exception != null)
```

Comment:
Would be great to make a helper extension method for this (if one doesn't exist already...I know I've seen similar logic in various places already)

Reply:
Sure, will follow up in a separate PR for this. Need to review we're logging errors from everywhere as well.

Reply:
Addressed